packages/block-library/src/embed/style.scss

```diff
@@ -1,74 +1,71 @@
-// Apply max-width to floated items that have no intrinsic width
-.block-editor-block-list__block[data-type="core/embed"][data-align="left"] .block-editor-block-list__block-edit,
-.block-editor-block-list__block[data-type="core/embed"][data-align="right"] .block-editor-block-list__block-edit,
-.wp-block-embed.alignleft,
-.wp-block-embed.alignright {
-	// Instagram widgets have a min-width of 326px, so go a bit beyond that.
-	max-width: 360px;
-	width: 100%;
-}
-
 .wp-block-embed {
 	// The embed block is in a `figure` element, and many themes zero this out.
 	// This rule explicitly sets it, to ensure at least some bottom-margin in the flow.
 	margin-bottom: 1em;
+
+	// don't allow iframe to overflow it's container.
+	iframe {
+		max-width: 100%;
+	}
+
+	// Apply max-width to floated items that have no intrinsic width.
+	&.alignleft,
+	&.alignright {
+		// Instagram widgets have a min-width of 326px, so go a bit beyond that.
+		max-width: 360px;
+		width: 100%;
+	}
 }
 
-.wp-embed-responsive {
-	.wp-block-embed {
-		// Add responsiveness to common aspect ratios.
-		&.wp-embed-aspect-21-9 .wp-block-embed__wrapper,
-		&.wp-embed-aspect-18-9 .wp-block-embed__wrapper,
-		&.wp-embed-aspect-16-9 .wp-block-embed__wrapper,
-		&.wp-embed-aspect-4-3 .wp-block-embed__wrapper,
-		&.wp-embed-aspect-1-1 .wp-block-embed__wrapper,
-		&.wp-embed-aspect-9-16 .wp-block-embed__wrapper,
-		&.wp-embed-aspect-1-2 .wp-block-embed__wrapper {
-			position: relative;
+.wp-block-embed__wrapper {
+	position: relative;
+}
 
-			&::before {
-				content: "";
-				display: block;
-				padding-top: 50%; // Default to 2:1 aspect ratio.
-			}
+// Add responsiveness to embeds with aspect ratios.
+.wp-embed-responsive .wp-has-aspect-ratio {
```

Comment:
To save some bytes, switched to `.wp-has-aspect-ratio` rather than listing all the different aspect-ratios. 
Any reason not to do this?